### ISSUE TITLE
The portrait foregrounds from AMD in Table.3.

### ISSUE THREAD

**Issue by ppf-hhh** [ZHKKKe/MODNet#103]
Thanks for sharing your awesome work! 
I would like to ask whether the portrait foregrounds used in Table 3 are only from the test set of the AMD dataset or are all portrait foregrounds in the training set and test set. 
Besides, when will your PPM-100 benchmark  be released?

**Comment by ZHKKKe** [comment]
Hi, thanks for your attention.
For your questions:
**Q1: I would like to ask whether the portrait foregrounds used in Table 3 are only from the test set of the AMD dataset or are all portrait foregrounds in the training set and test set.**
All portrait foregrounds in the training set and test set.

**Q2: When will your PPM-100 benchmark be released?**
We plan to release it in these two months. Sorry for the delay due to some reasons.

**Comment by ppf-hhh** [comment]
> Hi, thanks for your attention.
> For your questions:
> **Q1: I would like to ask whether the portrait foregrounds used in Table 3 are only from the test set of the AMD dataset or are all portrait foregrounds in the training set and test set.**
> All portrait foregrounds in the training set and test set.
> 
> **Q2: When will your PPM-100 benchmark be released?**
> We plan to release it in these two months. Sorry for the delay due to some reasons.

The bgs are from OpenImage dataset? I use images from COCO dataset (exclude images containing person) as backgrounds. And each foreground corresponds to 10 backgrounds. The mse and mad are 0.02 and 0.03.  Could you give me some suggestions?

**The code:
`
  import os
  import numpy as np
  import scipy
  import cv2
  from skimage.measure import label
  import argparse
  from scipy import ndimage, misc
  
  def gauss(x, sigma):
      y = np.exp(-x ** 2 / (2 * sigma ** 2)) / (sigma * np.sqrt(2 * np.pi))
      return y
  
  
  def dgauss(x, sigma):
      y = -x * gauss(x, sigma) / (sigma ** 2)
      return y
  
  def compute_mse(pred, gt):
      error_map = (pred-gt)**2
      return np.mean(error_map)
  
  def comput_mad(pred, gt):
      error_map = np.abs((pred - gt))
      return np.mean(error_map)
      
  def gaussgradient(im, sigma):
      epsilon = 1e-2
      halfsize = np.ceil(sigma * np.sqrt(-2 * np.log(np.sqrt(2 * np.pi) * sigma * epsilon))).astype(np.int32)
      size = 2 * halfsize + 1
      hx = np.zeros((size, size))
      for i in range(0, size):
          for j in range(0, size):
              u = [i - halfsize, j - halfsize]
              hx[i, j] = gauss(u[0], sigma) * dgauss(u[1], sigma)
  
      hx = hx / np.sqrt(np.sum(np.abs(hx) * np.abs(hx)))
      hy = hx.transpose()
  
      gx = scipy.ndimage.convolve(im, hx, mode='nearest')
      gy = scipy.ndimage.convolve(im, hy, mode='nearest')
  
      return gx, gy
  
  
  def compute_gradient_error(pred, gt):
  
      pred_x, pred_y = gaussgradient(pred, 1.4)
      gt_x, gt_y = gaussgradient(gt, 1.4)
  
      pred_amp = np.sqrt(pred_x ** 2 + pred_y ** 2)
      gt_amp = np.sqrt(gt_x ** 2 + gt_y ** 2)
  
      error_map = (pred_amp - gt_amp) ** 2
  
      return np.mean(error_map)
  
  def getLargestCC(segmentation):
      labels = label(segmentation, connectivity=1)
      largestCC = labels == np.argmax(np.bincount(labels.flat))
      return largestCC
  
  def compute_connectivity_error(pred, gt, step):
      
      h, w = pred.shape
  
      thresh_steps = list(np.arange(0, 1 + step, step))
      l_map = np.ones_like(pred, dtype=np.float) * -1
      for i in range(1, len(thresh_steps)):
          pred_alpha_thresh = (pred >= thresh_steps[i]).astype(np.int)
          gt_alpha_thresh = (gt >= thresh_steps[i]).astype(np.int)
  
          omega = getLargestCC(pred_alpha_thresh * gt_alpha_thresh).astype(np.int)
          flag = ((l_map == -1) & (omega == 0)).astype(np.int)
          l_map[flag == 1] = thresh_steps[i - 1]
  
      l_map[l_map == -1] = 1
  
      pred_d = pred - l_map
      gt_d = gt - l_map
      pred_phi = 1 - pred_d * (pred_d >= 0.15).astype(np.int)
      gt_phi = 1 - gt_d * (gt_d >= 0.15).astype(np.int)
      return np.mean(np.abs(pred_phi - gt_phi))
  
  
  parser = argparse.ArgumentParser(description='Process some integers.')
  parser.add_argument('--pred', type=str, required=True, help='pred path')
  parser.add_argument('--gt', type=str, required=True, help='gt path')
  args = parser.parse_args()
  count = 0
  files = os.listdir(args.pred)
  mse = 0.0
  mad = 0.0
  conn = 0.0
  grad = 0.0
  
  for file in files:
      count = count + 1
      gt = cv2.imread(args.gt+file.split('.')[0]+'.png',cv2.IMREAD_GRAYSCALE)
      #pred = cv2.imread("../../Human_matting/MODNet/Distinctions_valid/"+file.split('.')[0]+'.png',cv2.IMREAD_GRAYSCALE)
      pred = cv2.imread(args.pred+file.split('.')[0]+'.png',cv2.IMREAD_GRAYSCALE)
      gt = gt/255.0
      pred = pred/255.0
      mse += compute_mse(pred, gt)
      mad += comput_mad(pred, gt)

  print(mse/count)
  print(mad/count)`

**Comment by ZHKKKe** [comment]
What is the input resolution during testing?

**Comment by ppf-hhh** [comment]
The `demo/image_matting/colab/inference.py` will resize the image to around 512 and then upsample the prediction to the original image size and save it. The original image size is about 1000×1000 resolution.

**Comment by ppf-hhh** [comment]
I would be very grateful if you can provide the AMD test set or the image merge code used in your paper.

**Comment by ppf-hhh** [comment]
> What is the input resolution during testing?

The demo/image_matting/colab/inference.py will resize the image to around 512 and then upsample the prediction to the original image size and save it. The original image size is about 1000×1000 resolution.
Could you provide the AMD test set or the image merge code used in your paper?
